src/Chirp.Infrastructure/Repositories/CheepRepository.cs

```diff
@@ -134,6 +134,38 @@ public async Task CreateAuthor(string authorName)
             await _dbContext.SaveChangesAsync(); // Persist the changes to the database
         }
         
+        public async Task DeleteCheep(int cheepId)
+        {
+            var cheep = await _dbContext.Cheeps.FindAsync(cheepId);
+            if (cheep != null)
+            {
+                _dbContext.Cheeps.Remove(cheep);
+                await _dbContext.SaveChangesAsync();
+            }
+        }
+        
+        public async Task DeleteCheepByAuthor(AuthorDTO Author)
+        {
+            var cheeps = await _dbContext.Cheeps
+                .Where(cheep => cheep.Author.Name == Author.Name)
+                .ToListAsync();
+            if (cheeps != null)
+            {
+                _dbContext.Cheeps.RemoveRange(cheeps);
+
+                // Retrieve the author by name and then remove them
+                var author = await _dbContext.Authors
+                    .FirstOrDefaultAsync(a => a.Name == Author.Name);
+
+                if (author != null)
+                {
+                    _dbContext.Authors.Remove(author);
+                }
```

Comment:
I am not sure why we are deleting an author in the method where we are just deleting all cheeps. Maybe this should be refactored to its own method to avoid methods doing more than 1 thing?

Reply:
I understand your reasoning. However I still think we should respect the [seperation of concern](https://www.geeksforgeeks.org/separation-of-concerns-soc/) principle we have learnt from class. Our code might develop in ways we can't think of right now.
It's your decision what to do, since it is not a fundamental problem in our program but just discussing the fundamental principles of programming.